pkg/controller/servicebindingrequest/annotations.go

```diff
@@ -0,0 +1,116 @@
+package servicebindingrequest
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/dynamic"
+
+	"github.com/redhat-developer/service-binding-operator/pkg/apis/apps/v1alpha1"
+)
+
+const (
+	sbrNamespaceAnnotation = "service-binding-operator.apps.openshift.io/binding-namespace"
+	sbrNameAnnotation      = "service-binding-operator.apps.openshift.io/binding-name"
+)
+
+// extractNamespacedName returns a types.NamespacedName if the required service binding request keys
+// are present in the given data
+func extractNamespacedName(data map[string]string) types.NamespacedName {
+	namespacedName := types.NamespacedName{}
+	ns, exists := data[sbrNamespaceAnnotation]
+	if !exists {
+		return namespacedName
+	}
+	name, exists := data[sbrNameAnnotation]
+	if !exists {
+		return namespacedName
+	}
+	namespacedName.Namespace = ns
+	namespacedName.Name = name
+	return namespacedName
+}
+
+// GetSBRNamespacedNameFromObject returns a types.NamespacedName if the required service binding
+// request annotations are present in the given runtime.Object, empty otherwise. When annotations are
+// not present, it checks if the object is an actual SBR, returning the details when positive. An
+// error can be returned in the case the object can't be decoded.
+func GetSBRNamespacedNameFromObject(obj runtime.Object) (types.NamespacedName, error) {
+	namespacedName := types.NamespacedName{}
+	data, err := runtime.DefaultUnstructuredConverter.ToUnstructured(obj)
+	if err != nil {
+		return namespacedName, err
+	}
+
+	u := &unstructured.Unstructured{Object: data}
+
+	namespacedName = extractNamespacedName(u.GetAnnotations())
+	logger := log.WithValues(
+		"Resource.GVK", u.GroupVersionKind(),
+		"Resource.Namespace", u.GetNamespace(),
+		"Resource.Name", u.GetName(),
+		"Extracted.NamespacedName", namespacedName.String(),
+	)
+	if !IsSBRNamespacedNameEmpty(namespacedName) {
+		logger.Info("Not able to define SBR namespaced-name based on annotations!")
+		return namespacedName, nil
+	}
+
+	if u.GroupVersionKind() != v1alpha1.SchemeGroupVersion.WithKind(ServiceBindingRequestKind) {
```

Comment:
For controllers everything is based in a GVK, therefore I found it logical to centralize this process in a single place that would give a SBR or nil.

Reply:
This is happening in lines [51 to 61](https://github.com/otaviof/service-binding-operator/blob/5e9d5b372bc58b5e8af010b6e497a2e2bccae8af/pkg/controller/servicebindingrequest/annotations.go#L51). So the first attempt is to look at annotations, and then it checks for an actual SBR object. 

> This function is named GetSBRNamespacedNameFromObject but this if condition doesn't do anything with annotations.

Reply:
Since we decided to have a single controller, we have a single stream of events coming to the controller, and thus the idea is to always identify a SBR in any event. It's either defined by annotations, which are more numerous and should update more frequently, and if not found in annotation, then it tries to identify an actual SBR object.

Reply:
@sbose78 I've split the watch of SBR and other resources, please consider.